### PR TITLE
fix(textbox): losing focus on PointerUp when inside a Popup on WASM

### LINKED ISSUE #18790: [WASM] TextBox does not properly keep the focus when in a Flyout/Popup
### Current behavior

Please see repo below: when upgrading from UNO 5.2 -> 5.3 the AutoSuggestBox is screwed up when clicking before starting to type: the selection logic no longer works. Same when updating the repo to .NET 9 and latest UNO 5.5.

[Minimal.zip](https://github.com/user-attachments/files/17743891/Minimal.zip)

@jeromelaban my team and I again spent a lot of time chasing down this regression. Together with the other regressions I reported earlier ([here ](https://github.com/unoplatform/uno/issues/18599)and [here](https://github.com/unoplatform/uno/issues/16928)) we're still forced to stay with UNO 5.1 (!). I'm happy to continue investing resources and time in chasing down UNO regressions, but I also would appreciate if the regressions would be taken care of in not too far future. Thanks in advance

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #18790

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_PopupFocus.xaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    mc:Ignorable="d"
+    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+
+	<StackPanel>
+		<Button Content="Click me">
+			<Button.Flyout>
+				<Flyout>
+					<StackPanel>
+						<TextBlock>
+							Repro Steps:
```
Comment (MartinZikmund) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/AutoSuggestBox_PopupFocus.xaml:17]: I think `InputInjector` now works on WASM in runtime tests - so theoretically this should be possible to turn into a runtime test as well, or not?

File: src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.cs [hunk 2]
```diff
@@ -593,6 +593,17 @@ public double HorizontalOffset
 			);
 		#endregion
 
+#if __CROSSRUNTIME__
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.cs:596]: `__WASM__`?

File: src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.MuxInternal.cs [hunk 3]
```diff
@@ -179,6 +179,13 @@ protected override void OnPointerReleased(PointerRoutedEventArgs args)
 
 		private static bool ScrollContentControl_SetFocusOnFlyoutLightDismissPopupByPointer(UIElement pScrollContentControl)
 		{
+#if __CROSSRUNTIME__
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/ScrollViewer/ScrollViewer.MuxInternal.cs:182]: `__WASM__`?

File: src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Uno.cs [hunk 4]
```diff
@@ -28,6 +28,12 @@ public bool IsPointerCaptureRequired
...
 				// We should not capture the pointer on WASM by default because it would prevent the user from scrolling through text on selection.
 				// See https://github.com/unoplatform/uno/pull/16982, https://issues.chromium.org/issues/344491566
-				false));
+				false, static (dO, args) =>
+				{
+					if (((TextBox)dO)._contentElement is ScrollViewer sv)
+					{
+						sv.DisableSetFocusOnPopupByPointer = !(bool)args.NewValue;
+					}
+				}));
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Uno.cs:37]: `#if __WASM__`?

**Review by MartinZikmund — APPROVED** [review]

**Review by dr1rrb — APPROVED** [review]

**Comment by jeromelaban** [comment]
This test is failing, this is quite odd, I've restarted the build to validate again:

image

**Pull request merged**